Source/Singulink.Globalization.Currency/SortedMoneySet.cs

```diff
@@ -0,0 +1,365 @@
+    }
+
+    /// <inheritdoc cref="IReadOnlyMoneySet.TryGetValue(Currency, out Money)"/>
+    public bool TryGetValue(Currency currency, out Money value)
+    {
+        EnsureCurrencyAllowed(currency, nameof(currency));
+
+        if (_amountLookup.TryGetValue(currency, out decimal amount))
+        {
+            value = new Money(amount, currency);
+            return true;
+        }
+
+        value = default;
+        return false;
+    }
+
+    /// <inheritdoc cref="IReadOnlyMoneySet.TryGetValue(string, out Money)"/>
+    public bool TryGetValue(string currencyCode, out Money value)
+    {
+        var currency = _registry[currencyCode];
+        return TryGetValue(currency, out value);
+    }
+
+    public bool TryGetAmount(string currencyCode, out decimal amount)
+    {
+        var currency = _registry[currencyCode];
+        return _amountLookup.TryGetValue(currency, out amount);
+    }
+
+    private void AddInternal(decimal amount, Currency currency)
+    {
+        if (_amountLookup.TryGetValue(currency, out decimal existingAmount))
+        {
+            _amountLookup[currency] = existingAmount + amount;
+        }
+        else
+        {
+            _amountLookup[currency] = amount;
+        }
+    }
+
+    private void AddRangeInternal(IEnumerable<Money> values, bool ensureCurrenciesInRegistry)
+    {
+        foreach (var value in values)
+        {
+            var currency = value.CurrencyOrDefault;
+
+            if (currency == null)
+                continue;
+
+            if (ensureCurrenciesInRegistry)
+                EnsureCurrencyAllowed(currency, nameof(values));
+
+            if (_amountLookup.TryGetValue(currency, out decimal existingAmount))
+                _amountLookup[currency] = existingAmount + value.Amount;
+            else
+                _amountLookup.Add(currency, value.Amount);
+        }
+    }
+
+    private void EnsureCurrencyAllowed(Currency currency, string paramName)
+    {
+        if (!_registry.Contains(currency))
+            Throw(currency, paramName);
+
+        static void Throw(Currency currency, string paramName)
+        {
+            throw new ArgumentException($"The currency '{currency}' is not present in the set's currency registry.", paramName);
+        }
+    }
+
+    /// <summary>
+    /// Returns a string representation of the money values this set contains.
+    /// </summary>
+    /// <param name="format">The format to use for each money value. See <see cref="Money.ToString(string?, IFormatProvider?)"/> for valid money formats.
+    /// Prepend the format with the <c>"!"</c> character to ignore zero amount values.</param>
+    /// <param name="formatProvider">The format provider that will be used to obtain number format information. This should be a <see cref="CultureInfo"/>
+    /// instance for formats that depend on the culture, otherwise the current culture is used.</param>
+    public string ToString(string? format, IFormatProvider? formatProvider = null)

```

Comment:
Either put this into a `#region String Formatting` here or move this to where it should be with other public methods

Reply:
Taken care of.

Reply:
Still not in the right spot - alphabetical order please. This class is also missing the base `ToString()` override.

Reply:
Fixed.